client/dashboard/sites/site-preview-links/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { SectionHeader } from '../../components/section-header';
+import SitePreviewLink from '../../components/site-preview-link';
+import { DotcomFeatures } from '../../data/constants';
+import { siteHasFeature } from '../../utils/site-features';
+import type { Site } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+interface SitePreviewLinkProps {
+	site: Site;
+	title?: string;
+	description?: string;
+}
+
+export default function SitePreviewLinks( { site, title, description }: SitePreviewLinkProps ) {
+	const { data: links = [] } = useQuery( sitePreviewLinksQuery( site.ID ) );
+	const createMutation = useMutation( sitePreviewLinkCreateMutation( site.ID ) );
+	const deleteMutation = useMutation( sitePreviewLinkDeleteMutation( site.ID ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	if ( ! siteHasFeature( site, DotcomFeatures.SITE_PREVIEW_LINKS ) ) {
+		return null;
+	}
+
+	const handleChange = ( { enabled }: { enabled?: boolean } ) => {
+		if ( enabled ) {
+			createMutation.mutate( undefined, {
+				onSuccess: () => {
+					createSuccessNotice( __( 'Preview link enabled.' ), { type: 'snackbar' } );
+				},
+				onError: () => {
+					createErrorNotice( __( 'Failed to enable preview link.' ), { type: 'snackbar' } );
+				},
+			} );
+		} else {
+			links?.forEach( ( { code } ) => {
+				deleteMutation.mutate( code, {
+					onSuccess: () => {
+						createSuccessNotice( __( 'Preview link disabled.' ), { type: 'snackbar' } );
+					},
+					onError: () => {
+						createErrorNotice( __( 'Failed to disable preview link.' ), { type: 'snackbar' } );
+					},
+				} );
+			} );
+		}
+	};
+
+	const renderContent = () => {
+		const isMutationPending = createMutation.isPending || deleteMutation.isPending;
+
+		const fields: Field< { enabled: boolean } >[] = [
+			{
+				id: 'enabled',
+				label: 'Enable share link',
+				Edit: ( { field, onChange, data, hideLabelFromVision } ) => {
+					const { id, label, getValue } = field;
+					return (
+						<ToggleControl
+							__nextHasNoMarginBottom
+							label={ hideLabelFromVision ? '' : label }
+							checked={ getValue( { item: data } ) }
+							disabled={ isMutationPending }
+							onChange={ () => onChange( { [ id ]: ! getValue( { item: data } ) } ) }
+						/>
+					);
+				},
+			},
+		];
+
+		const form = {
+			type: 'regular' as const,
+			fields: [ 'enabled' ],
+		};
+
+		const data = { enabled: links.length > 0 };
+
+		return (
+			<form>
+				<VStack spacing={ 4 }>
+					<DataForm< { enabled: boolean } >
+						data={ data }
+						fields={ fields }
+						form={ form }
+						onChange={ handleChange }
+					/>
+					{ links?.map( ( link ) => (
+						<SitePreviewLink
+							key={ link.code }
+							{ ...link }
+							siteUrl={ site.URL }
+							disabled={ isMutationPending }
+						/>
+					) ) }
+				</VStack>
+			</form>
+		);
+	};
+
+	if ( title && description ) {
```

Comment:
Seems that with the latest changes, the title and description is always passed? In that case this logic is no longer needed.

Reply:
It's a bit of preparation in case we want to backport this to an A4A flow, where we're showing the form inside a Modal (so still not always in the form of a Card)

https://github.com/Automattic/wp-calypso/blob/2ef27289ad5ab3a789d96ead51733812a23490cb/client/sites-dashboard/components/site-preview-modal.tsx#L41

Maybe I'm thinking too far 😄